src/uwtools/resources/jsonschema/make-hgrid.jsonschema

```diff
@@ -0,0 +1,301 @@
+{
+  "properties": {
+    "make_hgrid": {
+      "additionalProperties": false,
+      "properties": {
+        "config": {
+          "additionalProperties": false,
+          "allOf": [
+            {
+              "if": {
+                "properties": {
+                  "grid_type": {
+                    "const": "from_file"
+                  }
+                }
+              },
+              "then": {
+                "required": [
+                  "my_grid_file"
+                ]
+              }
+            },
+            {
+              "if": {
+                "properties": {
+                  "grid_type": {
+                    "enum": [
+                      "tripolar_grid",
+                      "regular_lonlat_grid"
+                    ]
+                  }
+                }
+              },
+              "then": {
+                "required": [
+                  "nxbnds",
+                  "nybnds",
+                  "xbnds",
+                  "ybnds"
+                ]
+              }
+            },
+            {
+              "if": {
+                "properties": {
+                  "grid_type": {
+                    "const": "simple_cartesian_grid"
+                  }
+                }
+              },
+              "then": {
+                "required": [
+                  "nxbnds",
+                  "nybnds",
+                  "xbnds",
+                  "ybnds",
+                  "simple_dx",
+                  "simple_dy"
+                ]
+              }
+            },
+            {
+              "if": {
+                "properties": {
+                  "grid_type": {
+                    "enum": [
+                      "f_plane_grid",
+                      "beta_plane_grid"
+                    ]
+                  }
+                }
+              },
+              "then": {
+                "required": [
+                  "f_plane_latitude"
+                ]
+              }
+            },
+            {
+              "if": {
+                "properties": {
+                  "grid_type": {
+                    "const": "gnomonic_ed"
+                  },
+                  "nest_grids": {
+                    "type": "integer"
+                  }
+                }
+              },
+              "then": {
+                "required": [
+                  "halo"
+                ]
+              }
+            },
+            {
+              "if": {
+                "properties": {
+                  "do_cube_transform": {
+                    "type": "boolean"
+                  },
+                  "do_schmidt": {
+                    "type": "boolean"
+                  }
+                }
+              },
+              "then": {
+                "required": [
+                  "stretch_factor",
+                  "target_lat",
+                  "target_lon"
+                ]
+              }
+            }
+          ],
```

Comment:
Should we add `"minProperties": 3`? We require `execution` and `run_dir`, so that's 2, but then would it ever make sense to run this program with zero arguments? If not, such a check might at least clue the user to the fact that they need to supply more configuration information.

Reply:
We could also just require Config here, since from_file is included there, though I'll have to tweak the tests

Reply:
That's an even better idea. There's really no point in running the program without configuring it, right?

Reply:
Hrm. When I require `config` and try to update the example to have `config: {"grid_type": "from_file", "my_grid_file": "/path/to/my_grid_file"}` it is failing the schema trying to require the last conditional entries `["stretch_factor", "target_lat", "target_lon"]` and I've had no success modifying it to stop this. And its weird because it seemed to work correctly in the schema test before!

Reply:
I checked out your branch, made `"config"` required in the schema, then added a `"config": {"grid_type": "from_file", "my_grid_file": "/some/file"}` entry in the `config` fixture in `test_make_hgrid.py`. I then get the error you note above. I notice that if I remove _only_ the last of the `allOf` items -- the one that requires `stretch_factor`, `target_lat`, and `target_lon` -- the errors disappear. Maybe it's something about this specific item, though I don't immediately see what it could be. I suppose it's not just that the _last_ item, because, of course, when this one is removed the previous one becomes the "last item", and it's not causing a problem. I don't see yet what's happening...

Reply:
So I added `allOf` to ensure the multi-parameter conditionals worked, but it may prove to be unnecessary - I can hope, at least

Reply:
Still not successful - I'm going to call in @NaureenBharwaniNOAA for help since we did all those crazy conditions in the rocoto schema. I thought that should be right, but we might have managed to get away with some `anyOf` trickery in the end

Reply:
OK, I have a theory. If I change the schema item to
```
{
  "if": {
    "properties": {
      "do_cube_transform": {
        "type": "boolean"
      },
      "do_schmidt": {
        "type": "boolean"
      }
    },
    "required": ["do_cube_transform", "do_schmidt"]
  },
  "then": {
    "required": [
      "stretch_factor",
      "target_lat",
      "target_lon"
    ]
  }
}
```
the test passes. I think the issue is that the `if` in the original version is always true because `do_cube_transform` and `do_schmidt` are optional in the min-schema being tested, because those items are optional. I believe the `if` would become false if one of these values were provided with a non-`boolean` value in the mini-`dict` being tested, because the min-schema is explicit about that fact. The update to the min-schema above makes the _presence_ of those keys mandatory in the hypothetical scenario being tested.

The other min-schemas in the `allOf` sequence are different because they test not the _presence_ of their keys, but specific `const` or `enum` values.

Reply:
Confirmed and updated, thank you!